### PR TITLE
document breaking change policy

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/DataFrames.jl/issues/3259

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -227,8 +227,57 @@ all such objects are documented in this manual (in case some documentation is
...
+         that it was documented to handle - changes in handling of undocumented
+         arguments are not breaking;
+      2. the type of returned value by the funtion changes, but it still follows
+         the contract specified in the documentation; for example if a function
+         is documented to return a vector then changing its type from `Vector`
+         to `PulledVector` is not breaking;
+    * error behavior: code that threw an exception can change exception type
+      thrown or stop throwing an exception;
+    * changes in display (how objects are printed);
+    * changes to the satate of global objets from Base Julia whose state
```
Comment (jariji) [inline docs/src/index.md:253]: s/satate/state/

Comment (jariji) [reply docs/src/index.md:253]: s/objets/objects/

File: docs/src/index.md [hunk 2]
```diff
@@ -227,8 +227,57 @@ all such objects are documented in this manual (in case some documentation is
...
+      normally is considered volatile (e.g. state of global random number
+      generator).
+
+    All types and functions that are part of public API are guaranteed to go
+    through a deprecation period before a breaking change is made to them
+    changed or they would be removed.
```
Comment (jariji) [inline docs/src/index.md:259]: "changed"?

File: docs/src/index.md [hunk 3]
```diff
@@ -227,8 +227,57 @@ all such objects are documented in this manual (in case some documentation is
...
+
+    As a standard practice the change is implemented when a major release of
+    DataFrames.jl is made (e.g. functionalities deprecated in 1.x release
+    would be changed in 2.0 release).
+
+    In rare cases a breaking change might be introduced in a minor release.
+    In such case the changed behavior still goes through one minor release
```
Comment (jariji) [inline docs/src/index.md:266]: "such a case" or "such cases"

File: docs/src/index.md [hunk 4]
```diff
@@ -227,8 +227,57 @@ all such objects are documented in this manual (in case some documentation is
...
+    during which it is deprecated. The situations where such breaking change
```
Comment (jariji) [inline docs/src/index.md:267]: "such a breaking change" or "such breaking changes"

**Comment by bkamins** [comment]
@jariji - thank you!

File: docs/src/index.md [hunk 5]
```diff
@@ -227,8 +227,57 @@ all such objects are documented in this manual (in case some documentation is
...
+
+    Breaking changes to public and documented API are avoided in
+    DataFrames.jl where possible.
+
+    The following changes are not considered breaking:
+
+    * specific floating point values computed by operations may change at any
+      time; users should rely only on approximate accuracy;
+    * in functions that use the default random number generator provided by the
+      Base Julia the specific random numbers computed may change at any time;
```
Comment (nalimilan) [inline docs/src/index.md:240]:
```suggestion
    * in functions that use the default random number generator provided by
      Base Julia the specific random numbers computed may change across Julia versions;
```

File: docs/src/index.md [hunk 6]
```diff
@@ -227,8 +227,57 @@ all such objects are documented in this manual (in case some documentation is
...
+    * if the changed functionality is classified as a bug;
+    * if the changed behavior was not documented; two major cases are:
+      1. in its implementation some function accepted a wider range of arguments
+         that it was documented to handle - changes in handling of undocumented
+         arguments are not breaking;
+      2. the type of returned value by the funtion changes, but it still follows
+         the contract specified in the documentation; for example if a function
+         is documented to return a vector then changing its type from `Vector`
+         to `PulledVector` is not breaking;
```
Comment (nalimilan) [inline docs/src/index.md:249]:
```suggestion
      2. the type of the value returned by a function changes, but it still follows
         the contract specified in the documentation; for example if a function
         is documented to return a vector then changing its type from `Vector`
         to `PooledVector` is not considered as breaking;
```

File: docs/src/index.md [hunk 7]
```diff
@@ -227,8 +227,57 @@ all such objects are documented in this manual (in case some documentation is
...
+
+    * specific floating point values computed by operations may change at any
+      time; users should rely only on approximate accuracy;
+    * in functions that use the default random number generator provided by the
+      Base Julia the specific random numbers computed may change at any time;
+    * if the changed functionality is classified as a bug;
+    * if the changed behavior was not documented; two major cases are:
+      1. in its implementation some function accepted a wider range of arguments
+         that it was documented to handle - changes in handling of undocumented
+         arguments are not breaking;
```
Comment (nalimilan) [inline docs/src/index.md:245]:
```suggestion
         arguments are not considered as breaking;
```

File: docs/src/index.md [hunk 8]
```diff
@@ -227,8 +227,57 @@ all such objects are documented in this manual (in case some documentation is
...
+      normally is considered volatile (e.g. state of global random number
+      generator).
+
+    All types and functions that are part of public API are guaranteed to go
+    through a deprecation period before a breaking change is made to them
+    or they would be removed.
+
+    As a standard practice the change is implemented when a major release of
+    DataFrames.jl is made (e.g. functionalities deprecated in 1.x release
+    would be changed in 2.0 release).
```
Comment (nalimilan) [inline docs/src/index.md:263]:
```suggestion
    The standard practice is that breaking changes are implemented when a major
    release of DataFrames.jl is made (e.g. functionalities deprecated in a 1.x release
    would be changed in the 2.0 release).
```

File: docs/src/index.md [hunk 9]
```diff
@@ -227,8 +227,57 @@ all such objects are documented in this manual (in case some documentation is
...
+    In rare cases a breaking change might be introduced in a minor release.
+    In such a case the changed behavior still goes through one minor release
+    during which it is deprecated. The situations where such a breaking change
+    might be allowed are (still such breaking changes will be avoided if
+    possible):
+    
+    * the affected functionality was previously clearly identified in the
+      documentation as possible to be changed (for example in DataFrames.jl 1.4
+      release propagation rules of `:note`-style metadata are documented as
+      possible to be changed in the future);
```
Comment (nalimilan) [inline docs/src/index.md:274]:
```suggestion
      documentation as being subject to changes (for example in DataFrames.jl 1.4
      release propagation rules of `:note`-style metadata are documented as such);
```

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**